src/tests/CMakeLists.txt

```diff
@@ -59,10 +59,13 @@ if(ENABLE_TESTING)
   list(APPEND test_cases test_dftengine)
   list(APPEND test_cases test_bsecoupling)
   list(APPEND test_cases test_rate_engine)
+  if(USE_CUDA)
+    list(APPEND test_cases test_cudapipeline)
```

Comment:
we should have a unit test for cudamatrix as well.

Reply:
done